pkg/client/interface.go

```diff
@@ -0,0 +1,63 @@
+//go:generate mockgen -destination=../../internal/mocks/mockclient/query_client_mock.go -package=mockclient . Dialer,Connection
```

Comment:
1. This file will be populated with all client interfaces, not just `EventsQueryClient` (see: #65 & #94).
2. It SHOULD be in a separate package from any implementations to avoid dependency cycles.